static/app/views/discover/resultsSearchQueryBuilder.tsx

```diff
@@ -0,0 +1,297 @@
+    dataset,
+    includeTransactions = true,
+    placeholder,
+  } = props;
+
+  const api = useApi();
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+  const tags = useTags();
+
+  const filteredTags = useMemo(() => {
+    return omitTags && omitTags.length > 0
+      ? omit(tags, omitTags, EXCLUDED_FILTER_KEYS)
+      : omit(tags, EXCLUDED_FILTER_KEYS);
+  }, [tags, omitTags]);
+
+  const placeholderText = useMemo(() => {
+    return placeholder ?? t('Search for events, users, tags, and more');
+  }, [placeholder]);
+  const measurements = useMemo(() => getMeasurements(), []);
+  const functionTags = useMemo(() => getFunctionTags(fields), [fields]);
+
+  // Returns array of tag values that substring match `query`; invokes `callback`
+  // with data when ready
+  const getEventFieldValues = useCallback(
+    async (tag, query): Promise<string[]> => {
+      const projectIdStrings = (projectIds as Readonly<number>[])?.map(String);
+
+      if (isAggregateField(tag.key) || isMeasurement(tag.key)) {
+        // We can't really auto suggest values for aggregate fields
+        // or measurements, so we simply don't
+        return Promise.resolve([]);
+      }
+
+      // device.class is stored as "numbers" in snuba, but we want to suggest high, medium,
+      // and low search filter values because discover maps device.class to these values.
+      if (isDeviceClass(tag.key)) {
+        return Promise.resolve(DEVICE_CLASS_TAG_VALUES);
+      }
+      const fetchPromise =
+        dataset === DiscoverDatasets.SPANS_INDEXED
+          ? fetchSpanFieldValues({
+              api,
+              endpointParams: normalizeDateTimeParams(selection.datetime),
+              orgSlug: organization.slug,
+              fieldKey: tag.key,
+              search: query,
+              projectIds: projectIdStrings,
+            })
+          : fetchTagValues({
+              api,
+              endpointParams: normalizeDateTimeParams(selection.datetime),
+              orgSlug: organization.slug,
+              tagKey: tag.key,
+              search: query,
+              projectIds: projectIdStrings,
+              // allows searching for tags on transactions as well
+              includeTransactions: includeTransactions,
+              // allows searching for tags on sessions as well
+              includeSessions: includeSessionTagsValues,
+              dataset: dataset ? DiscoverDatasetsToDatasetMap[dataset] : undefined,
+            });
+
+      try {
+        const results = await fetchPromise;
+        return results.filter(({name}) => defined(name)).map(({name}) => name);
+      } catch (error) {
+        throw new Error('Unable to fetch event field values');
+      }
+    },
+    [
+      api,
+      organization,
+      selection.datetime,
+      projectIds,
+      includeTransactions,
+      includeSessionTagsValues,
+      dataset,
+    ]
+  );
+
+  const getTagList: TagCollection = useMemo(() => {
```

Comment:
nit: `getTagList` sounds like a function but this is a object. Probably just call `filterKeys`?